src/Discovery/DiscoveryInterface.php

```diff
@@ -25,11 +25,15 @@ interface DiscoveryInterface {
    * @param string $mimetype
    *   Mime type for which to get the WOPI client URL.
    *   This refers to config entries in the discovery.xml file.
+   * @param 'view'|'edit'|'view_comment'|null $action
```

Comment:
string $action, there are many more in WOPI protocol, even if we don't have them here. We can list these as examples.

Reply:
There is the WOPI protocol, and then there is Collabora which documents that it only supports those 3.
But ok I make it string.

Reply:
Then one day they add one, and our documentation is wrong.